src/components/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,838 @@
+
+export interface TimeRange {
+  start: Date | number,
+  end: Date | number,
+  timePeriodsKey?: string | undefined
+}
+
+export interface TimePeriod {
+  key: string // unique identifier
+  display: string
+  timeframeText: string
+  timeframeLength(): string
+  start(): Date
+  end(): Date
+}
+
+export interface TimeFrameSection {
+  section: string
+  values: TimePeriod[]
+}
+
+export interface DateTimePickerState {
+  abbreviatedDisplay: string
+  fullRangeDisplay: string | undefined
+  hidePopover: boolean
+  selectedRange: TimeRange
+  selectedTimeframe: TimePeriod
+  tabName: string
+}
+
+export default defineComponent({
+  name: 'KDateTimePicker',
+  components: {
+    KButton,
+    KIcon,
+    KPop,
+    KSegmentedControl,
+    DatePicker,
+  },
+  props: {
+    icon: {
+      type: Boolean,
+      required: false,
+      default: true,
+    },
+    modelValue: {
+      type: Object as PropType<TimeRange>,
+      required: true,
+      validator: (value: TimeRange) => value instanceof Date ||
+        (value.start !== undefined && value.end !== undefined),
+    },
+    /**
+     * Upper bound for `v-calendar` dates, everything after this date will be disabled
+     */
+    maxDate: {
+      type: Date,
+      required: false,
+      default: null,
+    },
+    /**
+     * Lower bound for `v-calendar` dates, everything preceding this date will be disabled
+     */
+    minDate: {
+      type: Date,
+      required: false,
+      default: null,
+    },
+    /**
+     * Sets a custom interval for the Minute select dropdown
+     */
+    minuteIncrement: {
+      type: Number,
+      required: false,
+      default: 5,
+    },
+    /**
+     * Determines which `v-calendar` type to initialize.
+     * Three of the values (`date`, `time`, `dateTime`) are passed verbatim to `v-calendar`,
+     * whereas `relative` denotes a component instance made up solely of time frames.
+     */
+    mode: {
+      type: String,
+      required: true,
+      validator: (value: string) => {
+        return ['relative', 'date', 'time', 'dateTime'].includes(value)
+      },
+    },
+    /**
+     * Help text displayed as the default mesage inside the input field.
+     * When "Clear" is clicked, the input will revert to displaying this.
+     */
+    placeholder: {
+      type: String,
+      required: false,
+      default: 'Select a time range',
+    },
+    /**
+     * Determines whether the `v-calendar` will allow a single date/time,
+     * or a range of dates/times.
+     */
+    range: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+    /**
+     * A custom set of time frames to be displayed as selectable buttons.
+     * The `timeframeLength`, `start`, and `end` values are passed in as functions,
+     * allowing for on-the-fly date boundary creation.
+     */
+    timePeriods: {
+      type: Array as PropType<Array<TimeFrameSection>>,
```

Comment:
Just changing the syntax here so it matches all the other occurrences. 

```suggestion
      type: Array as PropType<TimeFrameSection[]>,
```